packages/flutter_tools/lib/fuchsia_executable.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:args/args.dart';
+
+import 'runner.dart' as runner;
+
+import 'src/artifacts.dart';
+import 'src/base/common.dart';
+import 'src/base/context.dart';
+import 'src/base/file_system.dart';
+import 'src/commands/attach.dart';
+import 'src/commands/devices.dart';
+import 'src/commands/shell_completion.dart';
+import 'src/fuchsia/fuchsia_sdk.dart';
+import 'src/runner/flutter_command.dart';
+
+ final ArgParser parser = ArgParser.allowAnything()
+  ..addOption('verbose', abbr: 'v')
+  ..addOption('help', abbr: 'h')
+  ..addOption(
+    'frontend-server',
+    help: 'The path to the frontend server snapshot.',
+  )
+  ..addOption(
+    'dart-sdk',
+    help: 'The path to the patched dart-sdk binary.',
+  )
+  ..addOption(
+    'ssh-config',
+    help: 'The path to the ssh configuration file.',
+  );
+
+/// Main entry point for fuchsia commands.
+///
+/// This function is intended to be used within the fuchsia source tree.
+Future<void> main(List<String> args) async {
+  final ArgResults results = parser.parse(args);
+  final bool verbose = results['verbose'];
+  final bool help = results['help'];
+  final bool verboseHelp = help && verbose;
+  final File dartSdk = fs.file(results['dart-sdk']);
+  final File frontendServer = fs.file(results['frontend-server']);
+  final File sshConfig = fs.file(results['ssh-config']);
+
+  if (!dartSdk.existsSync()) {
+    throwToolExit('--frontend-server is required: ${dartSdk.path} does not exist.');
```

Comment:
Swap `--frontend-server` and `--dart-sdk` between this and the next block

Reply:
D'oh